### PR TITLE
Release qr scanner for flatpak

### PR DESCRIPTION
**Pull request by YoussefLasheen**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: build-flatpak.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+
+# Convert the archive of the Flutter app to a Flatpak.
```
Comment (hfiguiere) [inline build-flatpak.sh:4]: this script is overly complicated and unmaintainable.

File: dev.lasheen.qr.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+modules:
+  - name: qr
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        url: https://github.com/YoussefLasheen/qr/releases/download/v1.0.0/qr-Linux-Portable.tar.gz
```
Comment (hfiguiere) [inline dev.lasheen.qr.yml:23]: you still build a flatpak for aarch64 with x86_64 binaries

File: dev.lasheen.qr.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        sha256: 8d4bd10e5ff14fba70e2fefff09b1fbf23d11dd53bfe2430e3173764bed6658d
+      - type: dir
```
Comment (hfiguiere) [inline dev.lasheen.qr.yml:25]: don't use `dir`. list the individual files.

Also the .desktop, appstream and icons should be shipped by upstream.

File: packaging/dev.lasheen.qr.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      <image>https://raw.githubusercontent.com/YoussefLasheen/qr/master/screenshots/screenshot-desktop-1.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/YoussefLasheen/qr/master/screenshots/screenshot-desktop-2.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/YoussefLasheen/qr/master/screenshots/screenshot-desktop-3.png</image>
+    </screenshot>
+  </screenshots>
+</component>
```
Comment (hfiguiere) [inline packaging/dev.lasheen.qr.metainfo.xml:35]: missing `release`, `url`, `content_rating`.

File: dev.lasheen.qr.yml (hunk 2 above)
Comment (YoussefLasheen) [reply dev.lasheen.qr.yml:31]: I added a flathub.json file to build for x86_64 only

File: dev.lasheen.qr.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  - --socket=pulseaudio
+  - --share=network
+modules:
+  - name: qr
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - 'mkdir -p qr'
+      - 'tar -xf qr-Linux-Portable.tar.gz -C qr'
```
Comment (hfiguiere) [inline dev.lasheen.qr.yml:21]: if you set the type `archive` you don't need that.

**Comment by hfiguiere** [comment]
bot, build dev.lasheen.qr

File: dev.lasheen.qr.yml (hunk 5 above)
Comment (YoussefLasheen) [reply dev.lasheen.qr.yml]: Thanks for the tip. I made some changes. Are they any good?

**Comment by YoussefLasheen** [comment]
Is there something I should be doing? @hfiguiere

**Comment by hfiguiere** [comment]
bot, build dev.lasheen.qr

File: dev.lasheen.qr.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+sdk: org.freedesktop.Sdk
+command: qr
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --socket=pulseaudio
+  - --share=network
```
Comment (muelli) [inline dev.lasheen.qr.yml:13]: what does a QR code generator need network access for?
Can you leave a note here that explains why you're requesting this permission?

Comment (YoussefLasheen) [reply dev.lasheen.qr.yml:13]: The app uses the network to check for latest updates. @muelli 
Also I thought this is a standard permission.

**Comment by YoussefLasheen** [comment]
I appreciate what you guys are trying to do here. But this is a much worse experience than publishing on snapcraft. It took me literally 5 minutes to publish my flutter app on there. Here I am waiting without an SLA with no clue what am I waiting for. This is just a small app with non standard permissions that I am test-publishing before I publish more complete apps. I really like flatpak and I don't even use snaps but I have multiple apps published there with hundereds of users already and non on flathub.

File: dev.lasheen.qr.yml (hunk 6 above)
Comment (hfiguiere) [reply dev.lasheen.qr.yml:13]: and what will it do with that information? It can't update itself.

Comment (hfiguiere) [reply dev.lasheen.qr.yml:13]:
```suggestion
```

No permission is *standard*. If you don't need it don't put it.

Comment (YoussefLasheen) [reply dev.lasheen.qr.yml:13]: I need it to notify the user to update. I mentioned this to explain why I didn't add an explanation for the permission.

**Comment by hfiguiere** [comment]
Maybe you should remember that we are mostly all volunteers, so if you think you should have a SLA for people that volunteer, maybe you should shift reality. And also the number of people that can actually approve the initial submission is slim, but chances are higher to be approved by following the reviews from **volunteers**.

Patience is the first requirement.

**Comment by YoussefLasheen** [comment]
Your work is much appreciated. I just wanted some kind of a process. Do I need to check this repo everyday?. And also a time range would be nice. Is it going to be a week or month.

File: dev.lasheen.qr.yml (hunk 6 above)
Comment (barthalion) [reply dev.lasheen.qr.yml:13]: Yeah, I don't think this is the best idea. This is not Mac/Windows where users need to re-download the app to get update or perform anything specific. On modern Linux distros, flatpak updates are applied automatically. It should be removed.

Comment (YoussefLasheen) [reply dev.lasheen.qr.yml]: Sure @barthalion

**Comment by barthalion** [comment]
bot, build dev.lasheen.qr

**Pull request closed without merging**